app/src/main/java/io/github/zadam/triliumsender/LoginActivity.kt

```diff
@@ -42,129 +41,146 @@ class LoginActivity : AppCompatActivity() {
 
-    /**
-     * Represents an asynchronous login/registration task used to authenticate
-     * the user.
-     */
-    inner class UserLoginTask internal constructor(private val triliumAddress: String, private val username: String, private val password: String) : AsyncTask<Void, Void, LoginResult>() {
+            val loginResult = doLogin(triliumAddress, username, password)
 
-        val TAG : String = "UserLoginTask"
+            if (loginResult.success) {
+                // Store the address and api token.
+                TriliumSettings(this@LoginActivity).save(triliumAddress, loginResult.token!!)
+                // Announce our success.
+                Toast.makeText(this@LoginActivity, "Trilium connection settings have been successfully configured.", Toast.LENGTH_LONG).show()
+                // End the activity.
+                finish()
+            } else {
+                if (loginResult.errorCode == R.string.error_network_error
+                        || loginResult.errorCode == R.string.error_unexpected_response) {
 
-        override fun doInBackground(vararg params: Void): LoginResult {
+                    triliumAddressEditText.error = getString(loginResult.errorCode)
+                    triliumAddressEditText.requestFocus()
+                } else if (loginResult.errorCode == R.string.error_incorrect_credentials) {
+                    passwordEditText.error = getString(loginResult.errorCode)
+                    passwordEditText.requestFocus()
+                } else {
+                    throw RuntimeException("Unknown code: " + loginResult.errorCode)
+                }
+            }
+        } // uiScope.launch {}
+    } // private fun AttemptLogin
 
+
+    /**
+     * A result from a login attempt.
+     */
+    inner class LoginResult(val success: Boolean, val errorCode: Int?,
+                            val token: String? = null)
+
+    /**
+     * Makes the actual login http request in the IO thread, to avoid blocking the UI thread.
+     *
+     * @return A loginResult object.
+     */
+    private suspend fun doLogin(triliumAddress: String, username: String, password: String): LoginResult {
+        return withContext(Dispatchers.IO) {
+            val tag = "UserLoginCoroutine"
             val client = OkHttpClient()
 
             val json = JSONObject()
             json.put("username", username)
             json.put("password", password)
 
-            val body = RequestBody.create(Utils.JSON, json.toString())
+            val body = json.toString().toRequestBody(Utils.JSON)
             val request = Request.Builder()
-                    .url(triliumAddress + "/api/sender/login")
+                    .url("$triliumAddress/api/sender/login")
                     .post(body)
                     .build()
 
-            val response: Response;
+            val response: Response
 
             try {
+                // In the Dispatchers.IO context, blocking http requests are allowed.
```

Comment:
This is an annoying aggressive behavior in this lint check- I suspect it's a misconfig in okhttp that's causing it to miss that this is fine to do in the IO context.